feat: Kafka audit logging docs (#1680)
aiven/aiven-docs: docs/products/kafka/concepts/audit-logging.md

---
title: Audit logging for Aiven for Apache Kafka®
sidebar_label: Audit logging
early: true
---

> Audit logging for Aiven for Apache Kafka® records authenticated Kafka client activity on your service.

Reviewer: - both authenticated and non-authenticated
- 
```suggestion
Audit logging for Aiven for Apache Kafka® records all Kafka client activities on your service.
```

Author: I removed “authenticated” as the qualifier and clarified that audit logging can include unauthenticated access attempts, while avoiding “all activities” because audit logging is aggregated and configurable.

Audit logs show which Kafka users were active and can show which Kafka operations
were allowed or denied, depending on your audit logging settings.

Use audit logs to support security reviews and compliance workflows.

## How audit logging works

After a Kafka client authenticates with your Aiven for Apache Kafka service, Aiven
checks requested operations and records whether they are allowed or denied. Audit
logging writes entries to the service logs with the `AUDIT:` prefix.

Each audit entry identifies the Kafka user, also called a principal. In audit logs,
the principal is the Kafka user account that a client authenticates with, for example

> `User:avnadmin`.

Reviewer: ```suggestion
`User:avnadmin` (the principal used by default in Aiven).

Audit logging also records client activity performed internally within your service.
- Some are authenticated, for example `User:.karapace_sr` (the principal used by Karapace Schema Registry for accessing the `_schemas` topic).
- Others are not yet authenticated, for example broker replication and availability checks fall under `User:ANONYMOUS`. ```

## Audit log examples

A `user_operations` entry lists the Kafka user, source IP address, when the user first
became active in the aggregation period, and each operation with its result:

```text
[2024-07-29 11:02:27,861] AUDIT: User:alice (/192.0.2.10) was active since 2024-07-29T10:57:28.048485122Z: Allow WRITE on TOPIC:orders, Allow READ on GROUP:order-consumers
```

A `user_activity` entry includes only the Kafka user, source IP address, and when the
user first became active:

```text
[2024-07-29 11:02:27,861] AUDIT: User:alice (/192.0.2.10) was active since 2024-07-29T10:57:28.048485122Z.
```

## Limitations

When using audit logs, note these limitations:

- **Audit entries do not include exact operation times.** Audit logging groups
  activity over an aggregation period, so an entry's timestamp is not the exact time
  of an individual operation.
- **Kafka audit logs do not show which Aiven account made changes using Aiven tools.**
  Kafka audit logs record only Kafka client activity. Operations performed in the
  Aiven Console, Aiven CLI, Aiven API, Aiven Provider for Terraform, or Kubernetes

>   operator appear under an internal Aiven service account, not the individual Aiven

Reviewer: actually it depends
```suggestion
  operator can appear under an internal Aiven service account, not the individual Aiven
```

For example when consuming/producing to Kafka via Karapace REST Proxy, the user issuing the HTTP request is also the one connecting to Kafka if [kafka_rest_authorization](https://aiven.io/docs/products/kafka/reference/advanced-params#kafka_rest_authorization) is set to `True`, otherwise the generic `.karapace_rest` user is used.

---
title: Audit logging for Aiven for Apache Kafka®
sidebar_label: Audit logging
---

> Audit logging for Aiven for Apache Kafka® records Kafka client activity on your service, including authenticated activity and unauthenticated access attempts.

Reviewer: ```suggestion
Audit logging for Aiven for Apache Kafka® records Kafka client activity on your service. They typically include the authentication user and the operations it was authorized or denied for during a period
```

Author: Thanks, this helped make the intro more specific. I am not sure “authentication user” and “operations it was authorized or denied for” are clear for end users. I would also avoid “typically” because it can feel vague in docs.

Does this capture what you intended?

Audit logging for Aiven for Apache Kafka® records Kafka client activity on your service. Audit logs show which Kafka users were active and can show which Kafka operations were allowed or denied during a configured time period.

> Audit logs show which Kafka users were active and can show which Kafka operations

Reviewer: ```suggestion
```

> were allowed or denied, depending on your audit logging settings.

Reviewer: ```suggestion
depending on your audit logging settings.
```
